packages/trader/src/Stores/Modules/Trading/trade-store.js

```diff
@@ -308,10 +309,23 @@ export default class TradeStore extends BaseStore {
 
     @action.bound
     async prepareTradeStore() {
-        // fallback to default currency if current logged-in client hasn't selected a currency yet
-        this.currency = this.root_store.client.currency || this.root_store.client.default_currency;
         this.initial_barriers = { barrier_1: this.barrier_1, barrier_2: this.barrier_2 };
         await when(() => !this.root_store.client.is_populating_account_list);
+
+        // waits for `website_status` in order to set `stake_default` for the selected currency
+        await WS.wait('website_status');
+        runInAction(() => {
```

Comment:
`runInAction` is not needed here, since the `processNewValuesAsync` is an action itself. We can always set the `currency = this.root_store.client.currency || this.root_store.client.default_currency` in the outer scope to keep it more readable.

Reply:
we need to call `processNewValuesAsync` so that it resets default stake on currency change - [here](https://github.com/binary-com/deriv-app/pull/2075/files#diff-ac84bdaae0a85981966f330c025656c3aaa66895a9376e738bb503d088594ec0R688)
And `processNewValuesAsync` is not an action 😄